extensions/resteasy-reactive/rest-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/JaxrsMethodsProcessor.java

```diff
@@ -19,7 +24,29 @@ public boolean test(MethodInfo method) {
                 if (method.declaringClass().hasDeclaredAnnotation(ResteasyReactiveDotNames.PATH)) {
                     return true;
                 }
+                if (testMethod(method)) {
+                    return true;
+                }
+
+                // also look at interfaces implemented by the method's declaringClass
+                for (Type interfaceType : method.declaringClass().interfaceTypes()) {
+                    ClassInfo interfaceInfo = index.getClassByName(interfaceType.name());
+                    if (interfaceInfo != null) {
+                        if (interfaceInfo.hasDeclaredAnnotation(ResteasyReactiveDotNames.PATH)) {
+                            return true;
+                        }
+                        MethodInfo overriddenMethodInfo = interfaceInfo.method(method.name(),
+                                method.parameterTypes().toArray(new Type[0]));
+                        if (overriddenMethodInfo != null && testMethod(overriddenMethodInfo)) {
+                            return true;
+                        }
+                    }
+                }
+
+                return false;
+            }
 
+            private boolean testMethod(MethodInfo method) {
```

Comment:
Could we rename it `isJaxrsResourceMethod()` or something similar?

Reply:
Sure, can do. @cescoffier Any other comments from your side?
Besides that, I was wondering about the "shortcut" that was already present in the original code: If there is a \@Path annotation on the class, **all** methods pass the test. But a JAX-RS could contain JAX-RS methods (annotated with \@GET etc.) and also some (un-annotated) "helper" methods that are used internally by the JAX-RS methods. Now, \@RunOnVirtualThread only makes sense on the JAX-RS methods and not on the helper methods, so I was wondering what the reasoning of the "shortcut" was and whether it is actually valid?
On the other hand, I don't know how "exact" this whole test needs to be. It seems to be checking method annotations only (and not class annotations) at this time anyways, so maybe it's "close enough" to prevent people from putting \@RunOnVirtualThread in absurd places?!

Reply:
I have renamed the method and rebased the branch